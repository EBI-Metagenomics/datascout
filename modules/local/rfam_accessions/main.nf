process RFAM_ACCESSIONS {

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/4f/4fe75ceed4e54cac511cb591843ec3fe0016b9d2f996fcbc639b2289f886e5de/data' :
        'community.wave.seqera.io/library/python_pip_pymysql:0b6be43d90920e61' }"
    label "process_low"

    tag "${meta}"
    
    input:
      tuple val(meta), path(tax_ranks), val(rank)
      val(rfam_db)

    output:
      tuple val(meta), path("${meta.id}_rfam_dir"), emit: rfam_results
      path("versions.yml"), emit: versions

    script:
    """
    mkdir -p ${meta.id}_rfam_dir
    rfam_accessions.py --tax_file ${tax_ranks} --output_dir "${meta.id}_rfam_dir" --rank ${rank} --config ${rfam_db}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
    \$( rfam_accessions.py --config ${rfam_db} --version 2>&1 )
      Python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """
}

// Get families from RFAM and write list to file
