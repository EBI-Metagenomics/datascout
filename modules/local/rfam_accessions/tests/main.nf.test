nextflow_process {

    name "Test Process RFAM_ACCESSIONS"
    script "modules/local/rfam_accessions/main.nf"
    process "RFAM_ACCESSIONS"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = tuple([id:"test_sample1", genome_id:"GCA_003719455.1", ena_tax:"default"], "$projectDir/tests/reference_databases/test_sample1_tax_ranks.tsv", "default")
                input[1] = "$projectDir/assets/rfam_db.conf"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.rfam_results,
                process.out.versions
            ).match()
        }

    }

}
