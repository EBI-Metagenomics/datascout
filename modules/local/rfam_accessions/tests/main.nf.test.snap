{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "test_sample1",
                        "genome_id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    [
                        "rfam_ids.txt:md5,8950f0d41ad4850ac6b4fbba80c7ae65"
                    ]
                ]
            ],
            [
                "versions.yml:md5,8ca24dc01e3c84c5efa832419923cb09"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-10T19:37:17.170832"
    }
}